examples/blob2d-outerloop/blob2d.cxx

```diff
@@ -0,0 +1,217 @@
+                     .withDefault(true);
+    sheath = options["sheath"].doc("Sheath closure").withDefault(true);
+
+    /***************Calculate the Parameters **********/
+
+    Omega_i = e * B0 / m_i;    // Cyclotron Frequency
+    c_s = sqrt(e * Te0 / m_i); // Bohm sound speed
+    rho_s = c_s / Omega_i;     // Bohm gyro-radius
+
+    output.write(
+        "\n\n\t----------Parameters: ------------ \n\tOmega_i = {:e} /s,\n\tc_s = "
+        "{:e} m/s,\n\trho_s = {:e} m\n",
+        Omega_i, c_s, rho_s);
+
+    // Calculate delta_*, blob size scaling
+    output.write("\tdelta_* = rho_s * (dn/n) * {:e} ",
+                 pow(L_par * L_par / (R_c * rho_s), 1. / 5));
+
+    /************ Create a solver for potential ********/
+
+    if (boussinesq) {
+      // BOUT.inp section "phiBoussinesq"
+      phiSolver = Laplacian::create(Options::getRoot()->getSection("phiBoussinesq"));
+    } else {
+      // BOUT.inp section "phiSolver"
+      phiSolver = Laplacian::create(Options::getRoot()->getSection("phiSolver"));
+    }
+    phi = 0.0; // Starting guess for first solve (if iterative)
+
+    /************ Tell BOUT++ what to solve ************/
+
+    SOLVE_FOR(n, vort);
+
+    // Output phi
+    SAVE_REPEAT(phi);
+    SAVE_ONCE(rho_s, c_s, Omega_i);
+
+    return 0;
+  }
+
+  int rhs(BoutReal UNUSED(t)) override {
+
+    // Run communications
+    ////////////////////////////////////////////////////////////////////////////
+    mesh->communicate(n, vort);
+
+    // Invert div(n grad(phi)) = grad(n) grad(phi) + n Delp_perp^2(phi) = omega
+    ////////////////////////////////////////////////////////////////////////////
+
+    if (!boussinesq) {
+      // Including full density in vorticit inversion
+      phiSolver->setCoefC(n); // Update the 'C' coefficient. See invert_laplace.hxx
+      phi = phiSolver->solve(vort / n, phi); // Use previous solution as guess
+    } else {
+      // Background density only (1 in normalised units)
+      phi = phiSolver->solve(vort);
+    }
+
+    mesh->communicate(phi);
+
+    // Create data accessors for fast inner loop
+    auto n_acc = FieldAccessor<>(n);
+    auto vort_acc = FieldAccessor<>(vort);
+    auto phi_acc = FieldAccessor<>(phi);
+
+    const auto& region = n.getRegion("RGN_NOBNDRY"); // Region object
+
+#if RUN_WITH_RAJA
+    auto indices = region.getIndices(); // A std::vector of Ind3D objects
+    Ind3D* ob_i = &(indices)[0];
+
+    RAJA::forall<EXEC_POL>(RAJA::RangeSegment(0, indices.size()), [=] RAJA_DEVICE(int id) {
+      int i = ob_i[id].ind;
+#else
+    BOUT_FOR(i, region) {
+#endif
+      ddt(n_acc)[i] = -bracket(phi_acc, n_acc, i) - 2 * DDZ(n_acc, i) * (rho_s / R_c)
+                      + D_n * Delp2(n_acc, i);
+
+      ddt(vort_acc)[i] = -bracket(phi_acc, vort_acc, i)
+                         + 2 * DDZ(n_acc, i) * (rho_s / R_c)
+                         + D_vort * Delp2(vort_acc, i);
+
+#if RUN_WITH_RAJA
+    });
+#else
+    }
+#endif
```

Comment:
Tricky... I think it should be possible to make this nicer. We would need to be able to pass a lambda function into an operator like `=` or `<<` so that `BOUT_FOR_RAJA` expanded into something like
```
RajaWrapper(/*... */) << [=] RAJA_DEVICE(int id)
```
which could be followed by braces like `BOUT_FOR` and without the closing `)` that RAJA's `forall` needs.